### PR TITLE
fpga_tool: enable nakedret check

### LINKED ISSUE #388: [linter] Enable nakedret check
in ${PROJECT_ROOT}/.golangci.yml

### PR DESCRIPTION
**Pull request by rojkov**
closes #388

### CODE REVIEW

**Comment by mythi** [comment]
@kad PTAL

**Comment by rojkov** [comment]
@kad Could you please elaborate?

Here is an example playground for illustration https://play.golang.org/p/nfU28ZZv1Sk

**Comment by kad** [comment]
> @kad Could you please elaborate?
This refactoring uncovered issue with shadowing of `err`. However, errors returned from `GetAPIVersion()` or `PortGetInfo()` should not be lost and should be reported.

**Comment by rojkov** [comment]
Ok. So in other words there are bugs in the original code. Will fix.

**Comment by rojkov** [comment]
@kad please check the new version reflects the initial intention.

File: cmd/fpga_tool/fpga_tool.go [hunk 1]
```diff
@@ -247,21 +247,34 @@ func printFpgaPort(f fpga.Port, quiet bool) (err error) {
...
+		return err
+	}
+
+	fmt.Printf("Port Regions                     : %d\n", pi.Regions)
+	for idx := uint32(0); idx < pi.Regions; idx++ {
+		if ri, err2 := f.PortGetRegionInfo(idx); err2 == nil {
+			fmt.Printf("Port Region (Index/Size/Offset)  : %d / %d / %d\n", ri.Index, ri.Size, ri.Offset)
+		} else {
+			// Keep iterating, but don't loose at least the last error.
+			err = err2
```
Comment (kad) [inline cmd/fpga_tool/fpga_tool.go:273]: I think we can return error immediately. If earlier ioctl call to driver fails, there is very minor chance that subsequent will succeed

Comment (rojkov) [reply cmd/fpga_tool/fpga_tool.go]: Yeap, updated to return immediately.

**Comment by mythi** [comment]
@kad merging requires your approval

**Pull request merged**